apps/opik-backend/src/main/java/com/comet/opik/domain/SpanService.java

```diff
@@ -290,6 +291,11 @@ private List<Span> bindSpanToProjectAndId(SpanBatch batch, List<Project> project
                     String projectName = WorkspaceUtils.getProjectName(span.projectName());
                     Project project = projectPerName.get(projectName);
 
+                    if (project == null) {
+                        log.warn("Project not found for span project '{}' and default '{}'", span.projectName(), projectName);
+                        throw new NotFoundException("Project not found: %s".formatted(span.projectName()));
```

Comment:
This should never be possible, as project is created. Can you develop more how this part of the code can be reached?

Can you maybe add a test to prove the behaviour?

Reply:
To be honest, that is what I'm trying to understand. In which scenario this happens.  This is the exception:

```
java.lang.NullPointerException: Cannot invoke "com.comet.opik.api.Project.id()" because "project" is null
	at com.comet.opik.domain.SpanService.lambda$bindSpanToProjectAndId$25(SpanService.java:296)
        ...
	at com.comet.opik.domain.SpanService.bindSpanToProjectAndId(SpanService.java:298)
	at com.comet.opik.domain.SpanService.lambda$create$24(SpanService.java:276)
	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onNext(FluxMapFuseable.java:113)
	at io.opentelemetry.javaagent.shaded.instrumentation.reactor.v3_1.TracingSubscriber.onNext(TracingSubsc
```

Reply:
In that case, maybe just add the logs to see if it ever happens, but don't throw the exception.

Reply:
Ok, but it happens. We can check the error logs

Reply:
My favourite option would be removing the thrown `IllegalStateException` as in theory, it's unreachable code which can't be reproduced from tests. Although it happens in practice. But I can approve in the meantime.

Reply:
I will merge and raise a following PR